feat: Add Wallet Attestation request flow (#233)
italia/eidas-it-wallet-docs: docs/en/defined-terms.rst

     - All public and/or private entities, conforming to a technical profile and accredited by the Federation Authority, that provide citizens with an IT Wallet Instance.
   * - Wallet Attestation
     - Verifiable Attestation, issued by the Wallet Provider, that proves the security compliace of the Wallet Instance.
   * - Wallet Secure Cryptographic Device

>      - Hardware-backed secure environment for creating, storing, and/or managing cryptographic keys and data. Examples include Secure Elements (SE), Trusted Execution Environments (TEEs), and Strongbox.

Reviewer: the examples should include also other wscd technologies

Author: Ok thanks, I have included other technologies for the WSCD [here](https://github.com/italia/eudi-wallet-it-docs/pull/233/commits/c19cebfddbc66062b4099cec1f7ee89ede5b4d89).

   * - Credential Status Attestation
     - Verifiable Attestation proving that a related Digital Credential is not revoked.
   * - Device Integrity Service
     - A service provided by device manufacturers that verifies the integrity and authenticity of the app instance (Wallet Instance), as well as certifying the secure storage of private keys generated by the device within its dedicated hardware. It's important to note that the terminology used to describe this service varies among manufacturers.
   * - Wallet Hardware Keys

>      - During the app initialization, the Wallet Instance generates a pair of keys, one public and one private, which remain valid for the entire duration of the Wallet Instance's life. Functioning as a Master Key for the personal device, these Wallet Hardware Keys are confined to the OS domain and are not designed for signing arbitrary payloads. Their primary role is to provide a unique identification for each Wallet Instance.

Reviewer: too much tied on the local wscd, the Cryptographic Hardware Keys

WDYT about `Cryptographic Hardware Keys`?

Author: Great, renamed [here](https://github.com/italia/eudi-wallet-it-docs/pull/233/commits/dae1758cf47951f1cb44f9f8ac24ac9f3b0beb8c)

>    * - Wallet Hardware Key Tag

Reviewer: WDYT about `Cryptographic Hardware Key Tag`?

Author: Great, renamed [here](https://github.com/italia/eudi-wallet-it-docs/pull/233/commits/dae1758cf47951f1cb44f9f8ac24ac9f3b0beb8c)

     - A unique identifier created by the operating system for the Wallet Hardware Keys, utilized to gain access to the private key stored in the hardware.

>    * - Key Attestation

Reviewer: WDYT about `Cryptographic Hardware Key Tag`?

Author: Great, renamed [here](https://github.com/italia/eudi-wallet-it-docs/pull/233/commits/dae1758cf47951f1cb44f9f8ac24ac9f3b0beb8c)

     - All public and/or private entities, conforming to a technical profile and accredited by the Federation Authority, that provide citizens with an IT Wallet Instance.
   * - Wallet Attestation
     - Verifiable Attestation, issued by the Wallet Provider, that proves the security compliace of the Wallet Instance.
   * - Wallet Attestation Request

>      - Format of the request sent by the Wallet Instance to the backend Wallet Provider to obtain a Wallet Attestation.

Reviewer: ```suggestion
```

I would not put this in the defined terms because it ise defined in a dedicated section related to the flow

   * - Wallet Secure Cryptographic Device
     - Hardware-backed secure environment for creating, storing, and/or managing cryptographic keys and data. Examples include Secure Elements (SE), Trusted Execution Environments (TEEs), and Strongbox.
   * - Credential Status Attestation
     - Verifiable Attestation proving that a related Digital Credential is not revoked.
   * - Device Integrity Service

>      - Device manufacturer service that allows you to certify that the app instance (Wallet Instance) is intact and genuine and also allows you to certify that the private keys generated by the device are securely stored within the device's dedicated hardware. Please note that each manufacturer uses different terms to identify it.

Reviewer: ```suggestion
     - A service provided by device manufacturers that verifies the integrity and authenticity of the app instance (Wallet Instance), as well as certifying the secure storage of private keys generated by the device within its dedicated hardware. It's important to note that the terminology used to describe this service varies among manufacturers.
```

   * - Wallet Hardware Keys

>      - Pair of public and private keys generated by the Wallet Instance during app initialization. These keys remain valid throughout the entire lifespan of the Wallet Instance. Conceptually, the Wallet Hardware Keys serves as a personal device Master Key and, being part of the OS domain, cannot be used for signing any arbitrary payloads. The purpose of the Wallet Hardware Keys is to uniquely identify a Wallet Instance.

Reviewer: ```suggestion
     - During the app initialization, the Wallet Instance generates a pair of keys, one public and one private, which remain valid for the entire duration of the Wallet Instance's life. Functioning as a Master Key for the personal device, these Wallet Hardware Keys are confined to the OS domain and are not designed for signing arbitrary payloads. Their primary role is to provide a unique identification for each Wallet Instance.
```

   * - Wallet Hardware Key Tag

>      - Unique identifier of the Wallet Hardware Keys generated by the operating system and used to access the private key in the hardware.

Reviewer: ```suggestion
     - A unique identifier created by the operating system for the Wallet Hardware Keys, utilized to gain access to the private key stored in the hardware.
```

   * - Key Attestation

>      - Attestation signed by device OEM that gives you more confidence that the keys you use in your Wallet Instance are stored in a device's hardware-backed keystore

Reviewer: ```suggestion
     - An attestation from the device's OEM that enhances your confidence in the keys used in your Wallet Instance being securely stored within the device's hardware-backed keystore.
```

    - Application Programming Interface

>   * - **LoA**
>     - Level of Assurance

Reviewer 2: ```suggestion
  * - **LoA**
    - Level of Assurance
  * - **AAL**
    - Authenticator Assurance Level as defined in `<https://csrc.nist.gov/glossary/term/authenticator_assurance_level>`_
```
